store/sqlstore/channel_store.go

```diff
@@ -2854,6 +2855,30 @@ func (s SqlChannelStore) channelSearchQuery(term string, opts store.ChannelSearc
 		query = query.Where("c.Id NOT IN (SELECT ChannelId FROM GroupChannels WHERE GroupChannels.GroupId = ? AND GroupChannels.DeleteAt = 0)", opts.NotAssociatedToGroup)
 	}
 
+	if len(opts.TeamIds) > 0 {
+		query = query.Where(sq.Eq{"c.TeamId": opts.TeamIds})
+	}
+
+	if opts.GroupConstrained {
+		query = query.Where(sq.Eq{"c.GroupConstrained": true})
+	} else if opts.ExcludeGroupConstrained {
+		query = query.Where(sq.Or{
+			sq.NotEq{"c.GroupConstrained": true},
+			sq.Eq{"c.GroupConstrained": nil},
+		})
+	}
+
+	if opts.Public && opts.Private {
+		query = query.Where(sq.Or{
```

Comment:
nit: this is already done in the base query, no harm in it but a bit of repetition and maybe a future source of problems changing one and not the other. Probably I would remove that from the base query, and keep all the logic related to this here together, probably something like:

```go
if opts.Private && !opts.Public {
   ...filter by Private
} else if opts.Public && !opts.Private {
   ...filter by Public
} else {
   ...filter by Public or Private
}
```

Reply:
Done